src/store.rs

```diff
@@ -0,0 +1,196 @@
+use crate::job::{self, Job};
+use crate::workspace::Workspace;
+use anyhow::{Context, Result};
+use itertools::Itertools;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::{Component, Path, PathBuf};
+
+/// This struct manages all the levels of storage that we need in order to avoid
+/// doing as much work as possible. This mostly involves managing several layers
+/// of caches:
+#[derive(Debug)]
+pub struct Store {
+    root: PathBuf,
+
+    // This is stored as JSON for now to avoid taking another dependency,
+    // but it'd be good for it to be a real database (or database table)
+    // eventually. SQLite or Sled or something
+    inputs_to_content: HashMap<job::Id, String>,
+}
+
+impl Store {
+    pub fn new(root: PathBuf) -> Result<Self> {
+        let inputs_to_content = match File::open(&root.join("inputs_to_content.json")) {
+            Ok(file) => {
+                let reader = BufReader::new(file);
+                serde_json::from_reader(reader)
+                    .context("could not deserialize mapping from inputs to content")?
+            }
+            Err(err) if err.kind() == std::io::ErrorKind::NotFound => HashMap::default(),
```

Comment:
yeah, the first time we load the JSON file from disk it won't exist and we should just use an empty one and then persist it later.

This will go away later when we're using an actual DB (which I'd like to do ASAP, probably sled.)

Reply:
Ack